components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/MetadataNode.java

```diff
@@ -0,0 +1,133 @@
+package org.sonatype.nexus.repository.storage;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.repository.util.NestedAttributesMap;
+
+import com.orientechnologies.orient.core.id.ORID;
+import com.tinkerpop.blueprints.impls.orient.OrientVertex;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_ATTRIBUTES;
+import static org.sonatype.nexus.repository.storage.StorageFacet.P_LAST_UPDATED;
+
+/**
+ * Wraps an {@code OrientVertex} to provide a simpler API for working with stored component and asset metadata.
+ */
+public abstract class MetadataNode
+{
+  protected final OrientVertex vertex;
```

Comment:
Now that we're serving up collections of these, I think it's wise to implement equals() and hashCode(), based on referring to the same vertex.